### PR TITLE
endpoint: avoid allocating a lot of memory at first in topn

### PR DESCRIPTION
**Pull request by AndreMouche**
hi,
      If we order by c1 limit int.max, topn will alloc a lot of memory for heap, which leads to oom. This pr fix it.
       The related PR in tidb is https://github.com/pingcap/tidb/pull/3392

@BusyJay @andelf PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/server/coprocessor/endpoint.rs [hunk 1]
```diff
@@ -567,13 +567,19 @@ struct TopNHeap {
     err: Rc<RefCell<Option<String>>>,
 }
 
+const HEAP_DEFAULT_CAPACITY: usize = 1024;
```
Comment (hhkbp2) [inline src/server/coprocessor/endpoint.rs]: Try to use `xxx_MAX_CAPACITY_xxx`.

File: src/server/coprocessor/endpoint.rs [hunk 2]
```diff
@@ -567,13 +567,19 @@ struct TopNHeap {
...
+
 impl TopNHeap {
     fn new(limit: usize) -> Result<TopNHeap> {
         if limit == usize::MAX {
             return Err(box_err!("invalid limit"));
         }
+        let mut heap_capacity = HEAP_DEFAULT_CAPACITY;
```
Comment (andelf) [inline src/server/coprocessor/endpoint.rs]: cmp::min(.., ..)

**Comment by AndreMouche** [comment]
@hhkbp2 @andelf PTAL

**Comment by andelf** [comment]
LGTM

File: src/server/coprocessor/endpoint.rs [hunk 3]
```diff
@@ -567,13 +567,16 @@ struct TopNHeap {
     err: Rc<RefCell<Option<String>>>,
 }
 
+const HEAP_MAX_CAPACITY: usize = 1024;
```
Comment (zhangjinpeng1987) [inline src/server/coprocessor/endpoint.rs]: HEAP_INIT_CAPACITY

Comment (AndreMouche) [inline src/server/coprocessor/endpoint.rs]: I prefer to use the same name with `tidb` @zhangjinpeng1987

**Comment by hhkbp2** [comment]
LGTM

File: src/server/coprocessor/endpoint.rs (hunk 3 above)
Comment (zhangjinpeng1987) [inline src/server/coprocessor/endpoint.rs]: OK

**Pull request merged**